### ISSUE TITLE
No mapping found for [field_name] in order to sort on

### ISSUE THREAD

**Issue by adminpkowalewski** [Graylog2/graylog2-server#6378]
## Expected Behavior
Field is sorted

## Current Behavior
Recently upgraded to Graylog 3.1.0. I noticed that following new indexes have been added automatically:
1. gl-events
2. gl-system-events
Since then, when navigate to "Search" and try to sort any fields, I'm getting following error on ElasticSearch:

`[2019-09-03T12:51:57,751][DEBUG][o.e.a.s.TransportSearchAction] [mMVQnJP] [gl-events_5][0], node[mMVQnJPVTYO0N4l30esi6A], [P], s[STARTED], a[id=1Xfljj1dSKmAkgvDbxlqvw]: Failed to execute [SearchRequest{searchType=QUERY_THEN_FETCH, indices=[gl-events_6, graylog_26, gl-system-events_3, gl-events_5, gl-events_4, gl-system-events_4], indicesOptions=IndicesOptions[ignore_unavailable=false, allow_no_indices=true, expand_wildcards_open=true, expand_wildcards_closed=false, allow_aliases_to_multiple_indices=true, forbid_closed_indices=true, ignore_aliases=false, ignore_throttled=true], types=[message], routing='null', preference='null', requestCache=null, scroll=null, maxConcurrentShardRequests=5, batchedReduceSize=512, preFilterShardSize=64, allowPartialSearchResults=true, localClusterAlias=null, getOrCreateAbsoluteStartMillis=-1, source={"from":0,"size":150,"query":{"bool":{"must":[{"match_all":{"boost":1.0}}],"filter":[{"bool":{"must":[{"range":{"timestamp":{"from":"2019-09-03 11:46:57.748","to":"2019-09-03 11:51:57.748","include_lower":true,"include_upper":true,"boost":1.0}}}],"adjust_pure_negative":true,"boost":1.0}}],"adjust_pure_negative":true,"boost":1.0}},"sort":[{"ClientIP":{"order":"desc"}}]}}] lastShard [true]
org.elasticsearch.transport.RemoteTransportException: [mMVQnJP][127.0.0.1:9300][indices:data/read/search[phase/query]]
Caused by: org.elasticsearch.index.query.QueryShardException: No mapping found for [ClientIP] in order to sort on
        at org.elasticsearch.search.sort.FieldSortBuilder.build(FieldSortBuilder.java:321) ~[elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.sort.SortBuilder.buildSort(SortBuilder.java:153) ~[elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.SearchService.parseSource(SearchService.java:808) ~[elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.SearchService.createContext(SearchService.java:637) ~[elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:596) ~[elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:387) ~[elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.SearchService.access$100(SearchService.java:126) ~[elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:359) [elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:355) [elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.search.SearchService$4.doRun(SearchService.java:1107) [elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:41) [elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:751) [elasticsearch-6.8.2.jar:6.8.2]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-6.8.2.jar:6.8.2]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_222]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_222]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]`

## Possible Solution
Not sure

## Steps to Reproduce (for bugs)

1. Install Graylog 3.1.0
2. Feed Graylog with data (I use GELF)
3. Login to web GUI and press search button. Try to sort by any custom field

## Context
This issue has been reported by users that cannot sort fields anymore. Sorting works fine only if "All messages" stream is selected.

## Your Environment

* Graylog Version: 3.1.0
* Elasticsearch Version: 6.8.2
* MongoDB Version: v4.0.12
* Operating System:Ubuntu 18.04.3 LTS
* Browser version: Waterfox 56.2.13

**Comment by jalogisch** [comment]
From what Version did you upgrade - that isn't mention in your description. Cause if you are coming from 2.x you need to rotate your indices ONCE - what is written in the upgrade documentation. 

We are using GitHub issues for tracking bugs in Graylog itself, but this doesn't look like one. Please post this issue to our [discussion forum](https://community.graylog.org/) or join the [#graylog channel on freenode IRC](https://webchat.freenode.net/?channels=%23graylog).

Thank you!

**Comment by adminpkowalewski** [comment]
Thanks, Upgrade done from Graylog server 3.0.2.

When I browse "All messages" stream, then all sorting works fine. Issue appears only, when I use "Search" functionality....

However, I will test it on fresh install and raise it on forum you linked....

Regards!

**Comment by adminpkowalewski** [comment]
ok, I've tested it already on brand new installation:
1. Brand new Ubuntu 18.04.3 LTS 
2. Graylog server installed according to installation manual
http://docs.graylog.org/en/3.1/pages/installation/os/ubuntu.html

3.  I added GELF TCP input.
4. I sent data to Graylog.
5. I navigated to "Search", and tried to sort by field that was sent via GELF TCP input. I received following error:
"No mapping found for [EventID] in order to sort on"
6. Next I navigated to: "Streams"-> "All Messages" and I tried to sort by the same field (EventID), no issues then.

Would above be a bug or just regular issue?

Thanks,
Pawel
